views/account/profile/main.js

```diff
@@ -2,75 +2,228 @@ let live_alert_placeholder = document.getElementById("live-alert-placeholder");
 let accounts_table = document.getElementById("accounts-table-body");
 let add_new_label = document.getElementById("add-new-label");
 let add_new_value = document.getElementById("add-new-value");
+let add_new_type = document.getElementById("add-new-type");
 let add_new_account_submit = document.getElementById("add-new-account-submit");
 
-// These are dummy values for now; we would need to pull the casual and professional info from the db
-let labels = ["Instagram", "Facebook", "Email", "LinkedIn"];
-let values = [
-  "lelekaz",
-  "Leah Kazenmayer",
-  "kazenml1@tcnj.edu",
-  "https://www.linkedin.com/in/leah-kazenmayer/",
-];
+// Fetch the user_accounts from the database.
+let user_accounts = await (
+  await fetch("/account/profile/getall", { method: "GET" })
+).json();
 
-function add_row(table, label, value) {
+// Store the labels, values, and ids into separate arrays.
+let labels = [];
+let values = [];
+let types = [];
+let ids = [];
+if (user_accounts) {
+  user_accounts.forEach((account) => {
+    labels.push(account.label);
+    values.push(account.value);
+    types.push(account.type);
+    ids.push(account.id);
+  });
+}
+
+/**
+ * Adds a new row to the accounts table.
+ * @param {HTMLTableElement} table The table to add the row to.
+ * @param {string} label The label of the user_account.
+ * @param {string} value The value of the user_account.
+ * @param {number} id The ID of the user_account.
+ */
+function add_row(table, label, value, type, id) {
   let tr = document.createElement("tr");
   let label_td = document.createElement("td");
   let label_input = document.createElement("input");
   label_input.classList.add("form-control");
   let value_td = document.createElement("td");
   let value_input = document.createElement("input");
   value_input.classList.add("form-control");
-  let delete_td = document.createElement("td");
 
+  let type_td = document.createElement("td");
+  let type_btn = document.createElement("button");
+  type_btn.innerText = type == 0 ? "Casual" : "Professional";
+  type_btn.value = type;
+  type_btn.classList.add("btn", "btn-warning", "update", "btn-sm");
+
+  let delete_td = document.createElement("td");
   let delete_btn = document.createElement("button");
   delete_btn.innerText = "Delete";
   delete_btn.classList.add("btn", "btn-danger", "btn-sm");
 
   label_input.value = label;
+  label_input.oldvalue = label;
   label_td.appendChild(label_input);
   value_input.value = value;
+  value_input.oldvalue = value;
   value_td.appendChild(value_input);
 
+  type_td.appendChild(type_btn);
```

Comment:
I wonder if this is intuitive enough for the user to know that this is how you change the type. There should be some indicator to let them know that they can tap this to change it